storage_manager_new.py

```diff
@@ -5,184 +5,260 @@ class Storage:
-        self.name = name
-        self.option_1 = option_1
-        self.option_2 = option_2
-        self.option_3 = option_3
-        self.option_4 = option_4
-        self.option_5 = option_5
-
-    def menu_option_list(self):
-        option_list = [self.option_1, self.option_2,
-                       self.option_3, self.option_4, self.option_5]
-        return option_list
-
-    def start_program_interface(self, storage):
-        print('What would you like to do? Type a number.\n')
-        self.show_action_options_and_execute_user_choice(storage)
-
-    def show_action_options_and_execute_user_choice(self, storage):
-        menu = self.create_and_show_option_menu()
-        choice = Interface.get_user_choice(None)
+    option_dict = {1: 'Add item', 2: 'Show all items',
+                   3: 'Show list of items from a certain category or type', 4: 'Delete item or group of items', 5: 'Delete all items'}
+
+    def __init__(self, name, storage):
+        self.storage = storage
+        self.start_program_interface()
+
+    def start_process_of_adding_items(self):
+        while True:
+            new_item_string_data = self.collect_data(
+                '\nPlease provide new item data in order: '
+                'category, type, model, quantity (Example: Sound, Mixer, Midas M32, 1).\n')
+            self.break_if_data_is_menu(new_item_string_data)
+            adding_new_item = self.storage.add_item(new_item_string_data)
+            if adding_new_item == True:
+                self.inform_user('New item was added.')
+            else:
+                self.inform_user(
+                    'Adding item was cancelled because of an error, please try again.')
+                break
+            add_another_item = self.get_confirmation(
+                'Would you like to add another item?')
+            if add_another_item == False:
+                break
+
+    def start_process_of_showing_all_items(self):
+        print('')
+        print('----')
+        all_items_as_list = [str(item) for item in self.storage.items]
+        if all_items_as_list:
+            for item in sorted(all_items_as_list):
+                print(item)
+        else:
+            self.inform_user('List of items is empty.')
+        print('----')
+
+    def start_process_of_showing_items_by_category_or_type(self):
+        category_or_type = self.collect_data(
+            '\nPrint items from category or type:\n')
+        self.break_if_data_is_menu(category_or_type)
+        items_in_specified_category = self.storage.find_items_by_category_or_type(
+            category_or_type)
+        print('')
+        print('----')
+        if items_in_specified_category:
+            for item in items_in_specified_category:
+                print(item)
+        else:
+            self.inform_user(
+                f'Couldn\'t find any item from category/type {category_or_type}')
+        print('----')
+        print('')
+
+    def start_process_of_deleting_items(self):
```

Comment:
This method is huge and has lots of indentations (so logic cases etc.). This makes it not easy to read and understand. Any chance to split it into more smaller methods?

Reply:
How about adding comments, and creating a more logical order inside this method? Would that work, or splitting it into smaller methods is better idea?

Reply:
Answered outside of github but in biref, best idea is to split it into 9potentially "private") methods.